facade/src/main/java/org/jboss/pnc/facade/providers/ProductMilestoneProviderImpl.java

```diff
@@ -184,4 +214,96 @@ public void cancelMilestoneCloseProcess(String id)
 
         return queryForCollection(pageIndex, pageSize, sortingRsql, query, withProductVersionId(Integer.valueOf(productVersionId)));
     }
+
+    @Override
+    public Page<MilestoneInfo> getMilestonesOfArtifact(String id, int pageIndex, int pageSize) {
+        CriteriaBuilder cb = em.getCriteriaBuilder();
+
+        List<Integer> buildIn = getBuildIds(cb, id, Artifact_.buildRecords);
```

Comment:
You are getting milestone IDs and not IDs of the BuildRecords, the name of a variable is misleading.